### PR TITLE
Reader: add site filter to Automattic and P2s

### LINKED ISSUE #15343: Reader: Add P2 section
In the Reader, add a `Followed P2s` section. 

Ref. pbArwn-1q7-p2

The WordPress.com Reader:

Screen Shot 2020-11-19 at 1 38 32 PM

---
Tasks:
- [x] Reader: add P2 menu item and stream. [PR #15442]
- [x] Update post topic type from "team" to "organization". [PR #15449]
- [x] Update to use `organization_id` instead of `is_wpforteams_site`. (Lighthouse [issue](https://github.com/Automattic/p2/issues/1542), P2 [header](https://github.com/wordpress-mobile/WordPress-iOS/pull/15442) issue and post [detail](https://github.com/wordpress-mobile/WordPress-iOS/pull/15447#issuecomment-740269725) issue it should resolve). [PR #15461]
- [x] Universal Links: Add Reader P2 route to `Routes+Reader`. [PR #15455]
- [x] Add `wpios_reader_p2_shown` tracking. [PR #15462]
- [x] Update 'Following' site filter to display cached sites. [PR #15477]
- [x] Add site filter to P2 and A8C streams. [PRs #15484, #15494]
- [x] Site filter: update view title to match the stream name. [PR  #15497]

### PR DESCRIPTION
**Pull request by ScoutHarris**
Ref: #15343 

For Automattic and P2 reader streams, the 'Filter' tab bar option is now available to filter by site.

To test:

---
- Go to Reader > Following.
- Select `Filter`.
- Verify nothing has changed. That is, all followed sites and P2s appear in the `Sites` list, and filtering by `Topics` is available.

<kbd>following</kbd>

---
- Go to Reader > P2s.
- Select `Filter`.
- Verify the `Sites` list only contains non-A8C P2s. 
- Verify filtering by `Topic` is not available.

<kbd>p2</kbd>

---
- Go to Reader > Automattic.
- Select `Filter`.
- Verify the `Sites` list only contains A8C P2s. 
- Verify filtering by `Topic` is not available.

<kbd>a8c</kbd>

 ---
PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/Models/ReaderTeamTopic.swift [hunk 1]
```diff
@@ -11,6 +11,10 @@ import Foundation
         return slug == ReaderTeamTopic.a8cSlug ? .readerA8CShown : .readerP2Shown
     }
 
+    var organizationType: SiteOrganizationType {
```
Comment (aerych) [inline WordPress/Classes/Models/ReaderTeamTopic.swift:14]: Wfm.  However, if we wanted to have consistency with ReaderSiteTopic we could use the organization_id here also.  It's already returned by the /read/menu endpoint, we'd just need to add it to RemoteReaderTopic and then to the ReaderTeamTopic model. Not sure if it's worth the effort.  Up to you!

**Review by aerych — CHANGES_REQUESTED** [review]
Howdy @ScoutHarris 👋 
Looking good! I had one suggestion code wise but I'm not sure if it's worth the effort.
I spotted a couple of things while testing that we should look at. 
First, the title of the bottom sheet always says "Following".  Should it match the stream being filtered? 
Simulator Screen Shot - iPhone 11 Pro - 2020-12-14 at 15 14 31

Second, if I'm one of the filterable streams (e.g. P2s) and select a site, I can then choose another filterable stream (e.g. Following) but the filter still shows the title of the site I previously chose.  Should it reset to "Filter"? 
Simulator Screen Shot - iPhone 11 Pro - 2020-12-14 at 15 15 41

Back to you!

**Comment by ScoutHarris** [comment]
Hey @aerych .

>  the title of the bottom sheet always says "Following". Should it match the stream being filtered?

Should it? This is the view title, which to me is still accurate, i.e. every filter list is a list of Following. I can look into changing it if we like. Do we want to exactly match the stream name ("Following", "Automattic", "P2s", or do we want to append the specific stream name in some manner, or something similar ("Following", Following - Automattic", "Following - P2s")? cc @mattmiklic 

> the filter still shows the title of the site I previously chose. Should it reset to "Filter"?

Yes, yes it should. 🤦 I'll fix this.

File: WordPress/Classes/Models/ReaderTeamTopic.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/Models/ReaderTeamTopic.swift:14]: In this particular case, I don't think it's worth it. But it probably wouldn't hurt for consistency, clarity, etc. I'll do it in a follow up PR.

**Comment by ScoutHarris** [comment]
Hey @aerych .

I fixed the filter not resetting when the tabs change.

I'll follow up on these separately:
- 'Following' title (once we decide what it should be).
- Add `organization_id` to `ReaderTeamTopic`.

Ready for another review please.

**Review by aerych — APPROVED** [review]
Looks good!  
Sounds good on the follow up PR for the org id.
As far as the title... I had a moment of confusion thinking it was referring to the Following menu item and that it didn't update for the P2s menu item. I guess my expectation was it matched the label in the reader menu.  Curious what @mattmiklic thinks.  
Anyway, good to go!  :shipit: from me.

**Comment by ScoutHarris** [comment]
Thanks @aerych !

**Comment by mattmiklic** [comment]
I think updating the title of the sheet to match the view being shown probably makes sense. That said, I think it'd be good to remove the tab bar from the P2 and Automattic views, as when there's only one tab it's redundant.

**Comment by ScoutHarris** [comment]
Thanks @mattmiklic 

> I think updating the title of the sheet to match the view being shown probably makes sense. 

Just to clarify - to what? Match the stream name as Eric suggested?

> I think it'd be good to remove the tab bar from the P2 and Automattic views, as when there's only one tab it's redundant.

I see your point, but that might be a significant refactor. This view is very much designed to have a tab bar. 

Anyway, I'll update the view titles to match the stream name. I'll backburner removing the tab bar for now and consider exactly how to do that.

**Comment by mattmiklic** [comment]
> Just to clarify - to what? Match the stream name as Eric suggested?

Yep! And understood on the issues on removing the tab bar.

**Pull request merged**